netlify.toml

```diff
@@ -10,4 +10,5 @@ package = "netlify-plugin-checklinks"
   todoPatterns = [
     "content-type-missing", # I would like this to error local links and warn external, but here we are
     "https://github.com/numist/numi.st/blob/main/", # posts link to themselves on github, but those links point to `main` which doesn't work for pull requests
+    "<meta property=\"og:image\"", # meta links are absolute to the site's configured URL, not the actual deployed server
```

Comment:
I think the problem here is that `site.url` is hardcoded in `_config.yml` but it should actually be set based on the value of `URL` in [Netlify's build environment](https://docs.netlify.com/configure-builds/environment-variables/), when available. Fixing that would be much better than adding a URL validation exception.

Reply:
Yeah, the `<head>` contains a bunch of stuff that uses `site.url`:

``` html
<link rel="canonical" href="https://numi.st/">
<link type="application/atom+xml" rel="alternate" href="https://numi.st/feed.xml" title="numist log">
<meta property="twitter:image" content="https://numi.st/post/2022/travel-uke/IMG_1232.jpeg">
```

Probably the only reason this failure hasn't happened before is because I started using PRs after I'd already shipped a base version of the site.

But it also occurs to me that deployment previews probably also use `numi.st` for `URL`, and I'd have to use `DEPLOY_PRIME_URL`, but only when Netlify is deploying a non-`main` branch or the production website's links would stop being canonical? Argh.

Reply:
No wait deployment from the build root only happens after `netlify-plugin-checklinks` has completed successfully; I _could_ fix `site.url` but I'd still need to relativize all links (currently just seo and atom) for the checker to pass, and the seo ones are generated by a gem ☹️

Reply:
Yeah this is going to have to wait for another day. Seems possible I might be able to fix the "seo prepending site url instead of using a relative link" thing as part of jekyll/jekyll-seo-tag#461?

Reply:
I think you can just use the build option flag `--baseurl` instead of editing config.yml? https://jekyllrb.com/docs/configuration/options/

Reply:
I may have been too hasty to blame… it looks like `netlify-plugin-checklinks` actually _did_ notice the domain and resolved it to `/opt/build/repo/_site`? Also the file added for [another post](https://github.com/numist/numi.st/pull/10/files#diff-9adc99d51f58162a04d74c4298b6290980657e959060fc160c1d6e223ed23b80) seems to have checked clean.

```
✖ FAIL load _site/post/2022/reviving-a-yamaha-pss-a50/IMG_7584.jpeg
| operator: load
| expected: 200 _site/post/2022/reviving-a-yamaha-pss-a50/IMG_7584.jpeg
|   actual: ENOENT: no such file or directory, open '/opt/build/repo/_site/post/2022/reviving-a-yamaha-pss-a50/IMG_7584.jpeg'
|       at: _site/post/2022/reviving-a-Yamaha-PSS-A50/index.html:16:36 <meta property="og:image" content="https://numi.st/post/2022/reviving-a-yamaha-pss-a50/IMG_7584.jpeg">
```

Reply:
Oh wait shit I bet this is case-sensitivity again—`reviving-a-yamaha-pss-a50` should be `reviving-a-Yamaha-PSS-A50`